alf/environments/suite_carla.py

```diff
@@ -1076,6 +1118,56 @@ def _spawn_vehicles(self):
             self._other_vehicles) == num_vehicles, (
                 "Fail to create %s vehicles" % num_vehicles)
 
+    def _update_weather(self):
+        """Update the weather settings.
+
+        This function sample a new set of weather parameters once the
+        actual lasting time of the previous weather setting is up.
+        The actual lasting time of each weather setting is randomly sampled
+        from [0.25 * max_weather_length, max_weather_length].
+        After the termination of the previous weather setting, there is a
+        transition phase to linearly transit from the old to the new weather
+        settings.
+        """
+
+        # sample new weather parameter
+        if self._weather_length_count == 0:
+            # the actual lasting time of the new weather
+            self._actual_weather_length = max(
+                0.25, np.random.rand()) * self._max_weather_length
+            p = np.random.uniform(0, 100, len(WeatherParameters()))
+            weather = self._world.get_weather()
+            wp = WeatherParameters(*p)
+            prev_weather_parameter = [
+                weather.cloudiness, weather.precipitation,
+                weather.precipitation_deposits, weather.wind_intensity,
+                weather.fog_density, weather.fog_distance
+            ]
+
+            trans_steps = max(
+                1, self._actual_weather_length * self._weather_transition_ratio
+                / self._step_time)
+            self._dp = WeatherParameters(*(
```

Comment:
The requires the order at line 1141 is exactly same as the order at line 882, which is error prone.

Reply:
Good point. Improved the implementation and now we don't have to maintain the correspondence.